pkg/auth/providers/saml/saml_client.go

```diff
@@ -262,6 +295,51 @@ func (s *Provider) getSamlPrincipals(config *v32.SamlConfig, samlData map[string
 	return userPrincipal, groupPrincipals, nil
 }
 
+// FinalizeSamlLogout processes the logout obtained by the POST to /saml/slo from IdP
+func (s *Provider) FinalizeSamlLogout(w http.ResponseWriter, r *http.Request) {
+
+	if relayState := r.Form.Get("RelayState"); relayState != "" {
+		// delete the cookie
+		s.clientState.DeleteState(w, r, relayState)
+	}
+
+	redirectURL := s.clientState.GetState(r, "Rancher_FinalRedirectURL")
+
+	s.clientState.DeleteState(w, r, "Rancher_UserID")
+	s.clientState.DeleteState(w, r, "Rancher_Action")
+	s.clientState.DeleteState(w, r, "Rancher_FinalRedirectURL")
+
+	r.ParseForm()
+	err := s.serviceProvider.ValidateLogoutResponseRequest(r)
+	if err != nil {
+		log.Debugf("SAML [FinalizeSamlLogout]: response validation failed: %v", err)
+		if parseErr, ok := err.(*saml.InvalidResponseError); ok {
+			log.Debugf("SAML RESPONSE: ===\n%s\n===\nSAML NOW: %s\nSAML ERROR: %s",
+				parseErr.Response, parseErr.Now, parseErr.PrivateErr)
+		}
+
+		// assumes a redirect url with a `?...` already in place.
+		rURL, err := url.Parse(redirectURL)
+		...
```

Comment:
@pjbgf 
This needs fixup ... And having to deal with that I am now wondering how to handle an url parse error here. 
We will not have a proper `rURL`, only the `redirectURL`.
Which is likely bad.
I suspect at that point we have no choice but to go with string ops.
IIRC there might be a an encoder function for params we can use outside of URL methods ...
... https://pkg.go.dev/net/url#QueryEscape I believe.

Reply:
The latest commit has the proposed change.